spec/requests/posts_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
+
+RSpec.describe "Posts", type: :request do
+  let!(:user) { FactoryBot.create(:user) }
+  let!(:existing_post) { FactoryBot.create(:post, user: user) }
+
+  describe "GET /posts" do
+    it "returns a list of posts" do
+      get posts_path
+      expect(response).to have_http_status(302)
+    end
+  end
+
+  describe "POST /posts" do
+    it "saves a new Post" do
+      params = { post: { title: "New post", body: "New content.", user_id: user.id } }
+      expect {
+        post posts_path, params: { post: params }
+      }.to change(Post, :count).by(1)
+
+     expect(response).to redirect_to(post_path(Post.last))
```

Comment:
So this test was the one we had messaged about briefly and I believe I identified where the issue may be if you're still looking to solve it! It seems to be an issue with the authenticate_user! before_action that you call in the PostsController, meaning devise is not recognizing the user that you are creating with FactoryBot as an authorized user. I would double check that your factory is creating users with the correct attributes that match the validations that you've given that model and it should be able to get this test passing.